### PR TITLE
python3Packages.MDP: reenable tests

### PR DESCRIPTION
**Pull request by smancill**
###### Motivation for this change

Tests were disabled years ago by 140136ad956 (mdp: disable tests, 2017-09-14) because of missing dependencies that are not in nixpkgs, but current version 3.6 prints a warning and run the tests just fine.

    ============================= test session starts ==============================
    platform darwin -- Python 3.9.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
              python: 3.9.6.final.0
                 mdp: 3.6
     parallel python: NOT AVAILABLE: No module named 'pp'
              shogun: NOT AVAILABLE: No module named 'shogun'
              libsvm: NOT AVAILABLE: No module named 'libsvm'
              joblib: NOT AVAILABLE: No module named 'joblib'
             sklearn: NOT AVAILABLE: No module named 'scikits'
                numx: numpy 1.21.2
              symeig: symeig_fake
    Random Seed: 7710873

###### Things done

Add a custom `checkPhase` to run `pytest`.

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 140516` at 30453bbf run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.MDP</li>
    <li>python39Packages.MDP</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Do we have the missing deps in nixpkgs by now?

File: pkgs/development/python-modules/mdp/default.nix [hunk 1]
```diff
@@ -9,11 +15,27 @@ buildPythonPackage rec {
...
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+
+    pytest --seed 7710873 mdp
+    pytest --seed 7710873 bimdp
+
+    runHook postCheck
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mdp/default.nix:38]: Can we use pytestCheckHook instead?

Comment (smancill) [reply pkgs/development/python-modules/mdp/default.nix:38]: No, it fails if the hook is used. We need to set the `seed` option to run the tests, and because of that each directory must be run separately, or it will complain that the option is already set.

**Comment by smancill** [comment]
> Do we have the missing deps in nixpkgs by now?

joblib and scikit-learn are available in nixpkgs. If scikit-learn is used then the tests will find scipy and use it instead of numpy.
I have shogun ready as a Python module but I need to prepare the PR.

But then the tests take more time to run so I wasn't too convinced to add them.

File: pkgs/development/python-modules/mdp/default.nix [hunk 2]
```diff
@@ -9,11 +15,27 @@ buildPythonPackage rec {
     sha256 = "ac52a652ccbaed1857ff1209862f03bf9b06d093b12606fb410787da3aa65a0e";
   };
 
-  checkInputs = [ pytest ];
   propagatedBuildInputs = [ future numpy ];
 
-  # Tests disabled because of missing dependencies not in nix
-  doCheck = false;
+  checkInputs = [ pytest ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mdp/default.nix:20]: Please use pytestCheckHook.
